portfolio/src/main/webapp/contactMe.js

```diff
@@ -21,13 +17,57 @@ async function loadMessages(){
     return list;
 }
 
-function messageToHTML({ email, message, timestamp}){
-    let dateString = new Date(timestamp).toISOString();
+function messageToHTML({ id, email, message, timestamp}){
+    let dateString = new Date(timestamp).toDateString();
     return `
-    <li class="list-group-item p-3">
-        <h5>${ email }</h5>
-        <small class="text-disabled">${ dateString }</small>
-        <p>${ message }</p>
+    <li id="li-${ id }" class="list-group-item p-3 d-flex justify-content-between">
+        <div class="d-block">
+            <h5>${ email }</h5>
+            <small class="text-disabled">${ dateString }</small>
+            <p>${ message }</p>
+        </div>
+        <div class="d-block">
+            <button onclick="confirmDelete(${ id }, '${ email }')" class="btn btn-danger" data-bs-toggle="modal" data-bs-target="#confirmDeleteModal">
+                <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash-fill" viewBox="0 0 16 16">
+                    <path d="M2.5 1a1 1 0 0 0-1 1v1a1 1 0 0 0 1 1H3v9a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2V4h.5a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H10a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1H2.5zm3 4a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 .5-.5zM8 5a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7A.5.5 0 0 1 8 5zm3 .5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 1 0z"/>
+                </svg>
+            </button>
+        </div>
     </li>
     `;
 }
+
+function confirmDelete(id, email){
+    console.log('This is working hehge');
+    let span = document.getElementById('emailModal');
+    span.innerText = email;
+
+    let btn = document.getElementById('btnDelelte');
+    btn.setAttribute('onclick', `deleteMessage(${ id })`);
+}
+
+function deleteMessage(id){
+    let xhttp = new XMLHttpRequest();
```

Comment:
I believe you could still use the fetch method for this if you preferred, using:

fetch(`/contact-me?id=${ id }` {
  method: 'DELETE'
});

https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch

Reply:
Changed in the next commit!
I like the fetch method more.